src/AutorestSwift/Code Generator/ManagerConfig.swift

```diff
@@ -0,0 +1,160 @@
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Dispatch
+import Foundation
+import NIO
+
+class ManagerConfig: Encodable {
+    private var client: ChannelClient?
+    private var sessionId: String?
+
+    var inputFilename: String?
+    var outputDirectory: String?
+    var clearOutputFolder: Bool?
+    var projectDirectory: String?
+    var addCredential: Bool?
+    var credentialScopes: [String]?
+    var licenseHeader: String?
+    var namespace: String?
+    var tag: String?
+    var azureArm: Bool?
+    var headAsBoolean: Bool?
+    var title: String?
+    var description: String?
+    var clientSideValidation: Bool?
+    var packageName: String?
+    var packageVersion: String?
+    var trace: Bool?
+
+    enum CodingKeys: String, CodingKey {
+        case inputFilename, outputDirectory, clearOutputFolder, projectDirectory, addCredential, credentialScopes,
+            licenseHeader, namespace, tag, azureArm, headAsBoolean, title, description, clientSideValidation,
+            packageName, packageVersion, trace
+    }
+
+    /// Initialize from `CommandLineArguments` when running in standalone mode.
+    /// - Parameter args: `CommandLineArguments` object.
+    init() {
+        let args = CommandLineArguments()
+        inputFilename = args["--input-filename"]
+        outputDirectory = args["--output-directory"]
+        clearOutputFolder = Bool(args["--clear-output-folder"] ?? "False")
+        projectDirectory = args["--project-directory"]
+        addCredential = Bool(args["--add-credential"] ?? "False")
+        credentialScopes = args["--credential-scopes"]?.components(separatedBy: " ")
+        licenseHeader = args["--license-header"]
+        namespace = args["--namespace"]
+        tag = args["--tag"]
+        azureArm = Bool(args["--azure-arm"] ?? "False")
+        headAsBoolean = Bool(args["--head-as-boolean"] ?? "False")
+        title = args["--title"]
+        description = args["--description"]
+        clientSideValidation = Bool(args["--client-side-validation"] ?? "False")
+        packageName = args["--package-name"]
+        packageVersion = args["--package-version"]
+        trace = Bool(args["--trace"] ?? "False")
+    }
+
+    /// Initialize with RPC `ChannelClient` when running as an autorest plugin.
+    /// - Parameter client: RPC `ChannelClient` object.
+    /// - Parameter sessionId: RPC session ID
+    func loadValues(client: ChannelClient, sessionId: String, completion: @escaping () -> Void) {
+        self.client = client
+        self.sessionId = sessionId
+        // FIXME: Clearly, this is garbage, but I can't force these to behave synchronously
+        get(value: "input-filename") {
+            self.inputFilename = $0
+            self.get(value: "output-directory") {
+                self.outputDirectory = $0
+                self.get(value: "clear-output-folder") {
```

Comment:
Comment: This block is a bit hard to read. I understand each parameters need to be retrieve asynchronously.
I wonder if we can have an array of all the parameters we need to retrieve and maintains an index of where in the array that we have already retrieve the values. So in the call back of the get function, it get the next item of the array and call get again. If it reaches the end, call completion().

Reply:
I tried many ways to fix this ugly chaining and they all ended up in thread limbo. I'll give this a shot, though I may need to switch from attribute syntax to dictionary to make it work.